File: pdu/python/hako_msgs/pdu_cdr_conv_DisturbanceTemperature.py
from .pdu_pytype_DisturbanceTemperature import DisturbanceTemperature
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DisturbanceTemperature(writer: CdrWriter, src: DisturbanceTemperature):
    writer.write_float64(src.value)


def cdr_body_to_py_DisturbanceTemperature(reader: CdrReader, dst: DisturbanceTemperature):
    dst.value = reader.read_float64()
    return dst


def py_to_cdr_DisturbanceTemperature(src: DisturbanceTemperature) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DisturbanceTemperature(writer, src)
    return writer.bytes()


def cdr_to_py_DisturbanceTemperature(cdr_payload) -> DisturbanceTemperature:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DisturbanceTemperature()
    return cdr_body_to_py_DisturbanceTemperature(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_DisturbanceTemperature.js
import { DisturbanceTemperature } from './pdu_jstype_DisturbanceTemperature.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDisturbanceTemperatureConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DisturbanceTemperature } src
     */
    static to_cdr_body(writer, src) {
        writer.write_float64(src.value);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DisturbanceTemperature } dst
     * @returns { DisturbanceTemperature }
     */
    static cdr_body_to_js(reader, dst) {
        dst.value = reader.read_float64();
        return dst;
    }

    /**
     * @param { DisturbanceTemperature } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DisturbanceTemperature }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DisturbanceTemperature());
    }
}
